archive/imo/imo2008_q2.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2021 Manuel Candales. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Manuel Candales
+-/
+import data.real.basic
+
+/-!
+# IMO 2008 Q2
+(a) Prove that
+          ```x^2 / (x-1)^2 + y^2 / (y-1)^2 + z^2 / (z-1)^2 ≥ 1```
+for all real numbers x, y, z, each different from 1, and satisfying xyz = 1.
+
+(b) Prove that equality holds above for infinitely many triples of rational numbers x, y, z, each
+different from 1, and satisfying xyz = 1.
+
+# Solution
+(a) Since xyz = 1, we can apply the substitution x = a/b, y = b/c, z = c/a.
+Then we define m = c-b, n = b-a and rewrite the inequaility as LHS - 1 ≥ 0
+using c, m and n. We factor LHS - 1 as a square, which finishes the proof.
+-/
+
+lemma subst_abc (x y z : ℝ) (h : x*y*z = 1) :
+  ∃ a b c : ℝ, a ≠ 0 ∧ b ≠ 0 ∧ c ≠ 0 ∧ x = a/b ∧ y = b/c ∧ z = c /a :=
+begin
+  use x, use 1, use 1/y,
+  have h₁ : x ≠ 0               := by { intro p, rw p at h, simp at h, exact h },
```

Comment:
```suggestion
  have h₁ : x ≠ 0, { intro p, rw p at h, simp at h, exact h },
```
This is a minor suggestion, although in some cases it does become important: each `by` block has a certain amount of overhead, which is avoided with this style.

Reply:
No extra spaces either? Or is it ok to align the justifications?